### ISSUE TITLE
Number of members in public guilds page and guilds members list is wrong (+1) because deleted accounts remain in members list

### ISSUE THREAD

**Issue by habitrpguser** [HabitRPG/habitica#5847]
I'm in a few public guilds with less than 15 members (the maximum shown in the guild members list) - and they all show "1 more members" at the end of the guild members list (even if they have only 2 or 3 members), and also show that number of members (the listed ones plus one more) in the guilds list page. When I get the guild data with the Swagger UI API interface, though, there are only the members listed by name (and therefore one less than the number shown in Habitica).

I have no way to check if the numbers are also wrong for guilds with 15 members or more since Swagger UI only spits out the first 15 members.

This happens with almost all small public guilds no matter whether I'm a member or just peeking in. With some very few guilds it doesn't happen (for example "CIS", Group ID: 16c86941-33db-44f1-8e50-056c0d358b84), but I don't know why.

It doesn't happen with the private guilds and party that I'm a member of.

I haven't tested it in other browsers yet (seen it in Firefox) nor with restarting browser, clearing cache, etc. - but I have reloaded the pages.

**Comment by Alys** [comment]
This occurs when a player who is a member of the guild deletes their account without first leaving the guild. Their UUID remains in the list of guild members (and so it adds to the total count of members), but because it has no name associated with it, it doesn't appear in the visible list of members' names.

I checked several of the small public guilds, and interestingly in all of them the missing member was the same person (835ac530-23f0-460c-92ab-20ca0f977384). It looks like they might have joined a lot of guilds and then left. One of the small guilds I checked displayed "2 more members" and did have two deleted accounts in its members list (including 835ac530-23f0-460c-92ab-20ca0f977384).

I would guess that it's less likely to happen with private guilds because people who join them might be more engaged with Habitica (e.g., more friends here) and therefore are less likely to delete their accounts, and/or because 835ac530-23f0-460c-92ab-20ca0f977384 never received invitations. :-/

We'll fix this by purging member lists of deleted accounts after we fix [Issue #5253 - when player deletes HabitRPG account, they are not removed from party (and guilds?)](https://github.com/HabitRPG/habitrpg/issues/5253).
